.github/workflows/src/update-labels.js

```diff
@@ -103,6 +97,13 @@ export async function updateLabelsImpl({
   core.info(`labelsToAdd: ${JSON.stringify(labelsToAdd)}`);
   core.info(`labelsToRemove: ${JSON.stringify(labelsToRemove)}`);
 
+  if ((labelsToAdd.length > 0 || labelsToRemove.length > 0) && Number.isNaN(issue_number)) {
```

Comment:
```suggestion
  if ((labelsToAdd.length > 0 || labelsToRemove.length > 0) && !issue_number) {
```

Convention in JS is to use `!foo` to determine if a value is "falsy".  This includes things like `""`, `NaN`, `0`, `null`, and `undefined`.  Typically, you want to treat these all like "false", rather than specifically checking for `NaN`.